pkg/webhook/admission/defaulter_custom.go

```diff
@@ -21,29 +21,48 @@ import (
 	"encoding/json"
 	"errors"
 	"net/http"
+	"slices"
 
+	"gomodules.xyz/jsonpatch/v2"
 	admissionv1 "k8s.io/api/admission/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
 // CustomDefaulter defines functions for setting defaults on resources.
 type CustomDefaulter interface {
 	Default(ctx context.Context, obj runtime.Object) error
 }
 
+type defaulterOptions struct {
+	removeUnknownFields bool
+}
+
+type defaulterOption func(*defaulterOptions)
+
+// DefaulterRemoveUnknownFields makes the defaulter prune the fields that are not recognized in the local scheme.
```

Comment:
How about:
```suggestion
// DefaulterRemoveUnknownFields makes the defaulter prune fields that are in the json object retrieved by the webhook but not in the our go type. This happens for example when the CRD in the apiserver has fields that the our go type doens't know about, because its outdated.
```

Reply:
done

Reply:
Looks like you copied my grammar and spelling mistakes from the comment 😬 Lgtm other than that

Reply:
Yes, I'm pretty lazy that way :).
 Updated.